File: org.kevoree.modeling.genetic.framework/src/main/java/org/kevoree/modeling/genetic/api/PopulationFactory.java
package org.kevoree.modeling.genetic.api;

import org.kevoree.modeling.api.KMFContainer;
import org.kevoree.modeling.api.ModelCloner;

import java.util.List;

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 12/02/13
 * Time: 17:13
 */
public interface PopulationFactory<A extends KMFContainer> {

    public List<A> createPopulation();

    public ModelCloner getCloner();

}

File: org.kevoree.modeling.optimization.framework.core/src/main/java/org/kevoree/modeling/optimization/api/PopulationFactory.kt
package org.kevoree.modeling.optimization.api;

import org.kevoree.modeling.api.KMFContainer;
import org.kevoree.modeling.api.ModelCloner;

/**
 * Created with IntelliJ IDEA.
 * User: duke
 * Date: 12/02/13
 * Time: 17:13
 */
public trait PopulationFactory<A : KMFContainer> {

    public fun createPopulation() : List<A>;

    public fun getCloner() : ModelCloner;

}
